extensions/who-is-visiting-this-content/app.R

```diff
@@ -6,70 +6,197 @@ library(dplyr)
 library(glue)
 library(lubridate)
 library(tidyr)
+library(reactable)
+library(bsicons)
+library(ggplot2)
+library(plotly)
 
 shinyOptions(
   cache = cachem::cache_disk("./app_cache/cache/", max_age = 60 * 60 * 8)
 )
 
+extract_guid <- function(raw) {
+  guid_pattern <- "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+  match <- regmatches(raw, regexpr(guid_pattern, raw))
+  if (length(match) == 1) {
+    match
+  } else {
+    raw
+  }
+}
+
 source("get_usage.R")
 
-ui <- page_fillable(
-  useShinyjs(),
-  theme = bs_theme(version = 5),
-
-  card(
-    card_header("Who is Visiting This Content?"),
-    layout_sidebar(
-      sidebar = sidebar(
-        title = "Filters",
-        open = TRUE,
-
-        sliderInput(
-          "visit_lag_cutoff_slider",
-          label = "Visit Merge Window (sec)",
-          min = 0,
-          max = 600,
-          value = 1,
-          step = 0.5
+ui <- function(request) {
+  page_fluid(
+    useShinyjs(),
+    theme = bs_theme(version = 5),
+
+    div(
+      id = "guid_input_panel",
+      textInput("guid_field", "Paste in a content GUID or URL"),
+      actionButton("submit_guid", "Go", icon("arrow-right"))
+    ),
+
+    div(
+      id = "dashboard_panel",
+      style = "display:none;",
+      card(
+        card_header(
+          tags$div(
+            style = "display: flex; align-items: center; gap: 0.5rem;",
+            actionButton("clear_guid", "Back", icon("arrow-left"), class = "btn btn-sm"),
+            span("Content Detail")
+          )
         ),
+        layout_sidebar(
+          sidebar = sidebar(
+            title = "Controls",
+            open = TRUE,
+
+            selectInput(
+              "date_range_choice",
+              label = "Date Range",
+              choices = c("1 Week", "30 Days", "90 Days", "Custom"),
+              selected = "1 Week"
+            ),
+
+            conditionalPanel(
+              condition = "input.date_range_choice === 'Custom'",
+              dateRangeInput(
+                "date_range",
+                label = NULL,
+                start = today() - ddays(6),
+                end = today(),
+                max = today()
+              )
+            ),
+
+            sliderInput(
+              "visit_lag_cutoff_slider",
+              label = "Visit Merge Window (sec)",
+              min = 0,
+              max = 180,
+              value = 1,
+              step = 0.5
+            ),
+
+            textInput(
+              "visit_lag_cutoff_text",
+              label = NULL,
+              value = 1
+            ),
+
+            actionButton("clear_cache", "Clear Cache", icon = icon("refresh"))
```

Comment:
We can leave it in as we develop if it's useful, but is this something we would keep in the final version?

Reply:
I think it's mostly been useful during development, and could probably be removed in the final version.

(Part of me thinks, it'd be good to leave it in as a failsafe if we have a 24-hour window on the cache. That honestly speaks to a need to rethink the caching behavior more than keeping the button in, though. I honestly would love more experience with how performant the firehose is on servers with large usage databases.)

Reply:
> (Part of me thinks, it'd be good to leave it in as a failsafe if we have a 24-hour window on the cache. That honestly speaks to a need to rethink the caching behavior more than keeping the button in, though. I honestly would love more experience with how performant the firehose is on servers with large usage databases.)

💯 and also: the cache behavior when this is wrapped up in the larger list view / outerdashboard will need to be unified there too (e.g. blowing and regening the cache probably should be at that higher level and true for everything)